docs/guide/output-data-widgets.md

```diff
@@ -421,6 +421,80 @@ echo GridView::widget([
 ]);
 ```
 
+Most of the times using GridView filters in table headers is enough, but in case you need separate filter form,
+you can easily add it too. You can create partial view `_search.php` with the following contents:
+
+```php
+<?php
+
+use yii\helpers\Html;
+use yii\widgets\ActiveForm;
+
+/* @var $this yii\web\View */
+/* @var $model app\models\PostSearch */
+/* @var $form yii\widgets\ActiveForm */
+?>
+
+<div class="post-search">
+    <?php $form = ActiveForm::begin([
+        'action' => ['index'],
+        'method' => 'get',
+    ]); ?>
+
+    <?= $form->field($model, 'title') ?>
+
+    <?= $form->field($model, 'creation_date') ?>
+
+    <div class="form-group">
+        <?= Html::submitButton('Search', ['class' => 'btn btn-primary']) ?>
+        <?= Html::submitButton('Reset', ['class' => 'btn btn-default']) ?>
+    </div>
+
+    <?php ActiveForm::end(); ?>
+</div>
+```
+
+and include it in `index.php` view like so:
+
+```php
+<?= $this->render('_search', ['model' => $searchModel]) ?>
+```
+
+Note that if you use Gii to generate CRUD code, separate filter form (`_search.php`) is generated by default,
+but not displayed in `index.php` view, so you have to uncomment it if needed.
+
+Separate filter form is useful when you need to filter by date range for example.
```

Comment:
Maybe, we could show more generic case
`need to filter by date range for example` -> `need to filter by fields, that are not displayed in GridView, or for special filtering conditions, like date range.`